go/tendermint/apps/staking/state/state.go

```diff
@@ -525,6 +547,82 @@ func (s *MutableState) TransferFromCommon(ctx *abci.Context, toID signature.Publ
 	return ret, nil
 }
 
+// AddRewards computes and transfers the staking rewards to active escrow accounts.
+// If an error occurs, the pool and affected accounts are left in an invalid state.
+func (s *MutableState) AddRewards(time epochtime.EpochTime) error {
+	steps, err := s.RewardSchedule()
+	if err != nil {
+		return err
+	}
+	var stepStart epochtime.EpochTime
+	var activeStep *staking.RewardStep
+	for _, step := range steps {
+		if time < step.Until {
+			activeStep = &step
+			break
+		}
+		stepStart = step.Until
+	}
+	if activeStep == nil {
+		// We're past the end of the schedule.
+		return nil
+	}
+
+	stepRelativeTime := time - stepStart
+	if stepRelativeTime%activeStep.Interval != 0 {
+		return nil
+	}
+
+	commonPool, err := s.CommonPool()
+	if err != nil {
+		return errors.Wrap(err, "loading common pool")
+	}
+
+	s.Snapshot.IterateRange(accountKeyFmt.Encode(), nil, true, func(key, value []byte) bool {
```

Comment:
This seems to go over all accounts? Shouldn't we only reward accounts that belong to entities running validator nodes? Otherwise people can decide to delegate to escrow accounts that don't run any nodes and thus get all the rewards without taking on any risk (as those accounts will never be slashed).

Reply:
Maybe we should record all nodes that were active for larger part of an epoch and only distribute rewards to accounts belonging to their entities.

Reply:
ok I'll work on that

Reply:
went with distributing reward when an entity's node is considered for election